File: remote/src/main/java/com/kelsos/mbrc/interfaces/IEvent.java
package com.kelsos.mbrc.interfaces;

import com.kelsos.mbrc.constants.UserInputEventType;

/**
 * Interface that represents the an event.
 */
public interface IEvent {
  /**
   * Gets the type of the event.
   *
   * @return The string representing the type.
   */
  @UserInputEventType.Event String getType();
}

File: remote/src/main/kotlin/com/kelsos/mbrc/interfaces/IEvent.kt
package com.kelsos.mbrc.interfaces

import com.kelsos.mbrc.constants.UserInputEventType

/**
 * Interface that represents the an event.
 */
interface IEvent {
  /**
   * Gets the type of the event.

   * @return The string representing the type.
   */
  val type: String
}
